### DISCUSSION TITLE
Yanked versions

### DISCUSSION THREAD

**Discussion by Desidero** [apache/airflow#30114] [General] [answered]
Sometime in the last few days, a couple of older versions of Airflow were Yanked from pypi. I looked around to see if there was any published reasoning for it, but couldn't find anything.

Does anyone know why 1.10.11 and 1.10.13 were removed but 1.10.12 and a bunch of earlier versions remain?

**Comment by potiuk** [comment, marked as answer]
They were yanked years ago not recently because they had serious bugs. Details are explained in the message when you try to install them manually with==version (which is still possible for yanked version if you really want). 

Maybe you did not realize they were yanked because  similarly to old and already more then 2 years being end of life airflow version you also used some 4 years old pip version that did not have yanking support yet ?  That would be my guess. 

My suggestion to upgrade to software that is maintained Airflow 2.5 (1.10 is not maintained at all and you will have almost no-one using it so if you have any problems, you wil have to solve them on your own)

**Comment by notatallshaw-gts** [comment]
FYI what might have changed is that you updated Pip. As more recent versions of Pip respect that a package is Yanked much better than older versions, and therefore will complain in more circumstances and fail to install. You can always override this by exactly specifying the version, e.g. `apache-airflow==1.10.13`.

Fun fact: One of the issues with 1.10.13 is it was triggering the then new Pip resolver to download ancient versions setuptools that failed to install on new versions of Python, and even once Pip added a hack to resolve this it would never resolve the dependencies, both of these issues have recently been fixed on Pip main: https://github.com/pypa/pip/issues/11836
